sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallScheduler.java

```diff
@@ -73,16 +73,13 @@ private Plan getPlan() {
         }
 
         // create one UninstallStep per unique Resource, including Executor resources
-        List<Step> taskSteps = new ArrayList<>();
-        for (Protos.Resource resource : ResourceCollectionUtils.getAllResources(stateStore.fetchTasks())) {
-            Optional<String> resourceId = ResourceCollectionUtils.getResourceId(resource);
-            if (!resourceId.isPresent()) {
-                continue;
-            }
-            Status status = resourceId.get().startsWith(TOMBSTONE_MARKER) ? Status.COMPLETE : Status.PENDING;
-            taskSteps.add(new UninstallStep(resourceId.get(), status));
-        }
-
+        // We filter to unique Resource Id's, because Executor level resources are tracked
+        // on multiple Tasks.
+        List<Protos.Resource> allResources = ResourceCollectionUtils.getAllResources(stateStore.fetchTasks());
```

Comment:
@benclarkwood added a unit test `testInitialPlanTaskResourceOverlap` around that scenario. I'll add in your comments so the intent is clearer in the code.